packages/server/utils/atlassian/jiraImages.ts

```diff
@@ -0,0 +1,87 @@
+import cheerio from 'cheerio'
+import base64url from 'base64url'
+import crypto from 'crypto'
+import path from 'path'
+import ms from 'ms'
+import getRedis from '../getRedis'
+import AtlassianManager from 'parabol-client/utils/AtlassianManager'
+
+export const NO_IMAGE_BUFFER = Buffer.from('X')
+export const IMAGE_TTL_MS = ms('2h')
+
+const serverSecret = process.env.SERVER_SECRET
+if (!serverSecret) {
+  throw new Error('Missing SERVER_SECRET environment variable!')
+}
+
+/**
+ * Parses a JIRA issue description and replaces the image urls with Parabol's image urls
+ * where the image urls are hashed and look like this: `/jira-attachements/<hash>.[ext]`
+ * @param {string} descriptionHTML - HTML string of related Jira issue description
+ * @returns {UpdateJiraImagesResult} - Record of orginal image urls to hashed image urls
+ */
+export const updateJiraImageUrls = (cloudId: string, descriptionHTML: string) => {
+  const imageUrlToHash = {} as Record<string, string>
+  const projectBaseUrl = `https://api.atlassian.com/ex/jira/${cloudId}`
+
+  const $ = cheerio.load(descriptionHTML)
+  $('body')
+    .find('img')
+    .each((_i, img) => {
+      const imageUrl = $(img).attr('src')
+      if (!imageUrl) return
+
+      const absoluteImageUrl = `${projectBaseUrl}${imageUrl}`
+      const extname = path.extname(absoluteImageUrl)
+      const imageUrlWithoutExt = absoluteImageUrl.replace(extname, '')
```

Comment:
foo.png.png would fail here.
imageUrlWithoutExtension = absoluteImageUrl.slice(0, absoluteImageUrl.length - extname.length)

Reply:
💯  I def need to be less optimistic when coding!